packages/ipfs-message-port-server/src/core.js

```diff
@@ -0,0 +1,368 @@
+ * @typedef {import('./ipfs').FileInput} DecodedFileInput
+ */
+
+/**
+ * @template T
+ * @typedef {import('ipfs-message-port-protocol/src/core').RemoteCallback<T>} RemoteCallback
+ */
+
+/**
+ * @template T
+ * @typedef {import('ipfs-message-port-protocol/src/core').RemoteIterable<T>} RemoteIterable
+ */
+
+/**
+ * @typedef {Object} AddOptions
+ * @property {string} [chunker]
+ * @property {number} [cidVersion]
+ * @property {boolean} [enableShardingExperiment]
+ * @property {HashAlg} [hashAlg]
+ * @property {boolean} [onlyHash]
+ * @property {boolean} [pin]
+ * @property {RemoteCallback<number>|void} [progress]
+ * @property {boolean} [rawLeaves]
+ * @property {number} [shardSplitThreshold]
+ * @property {boolean} [trickle]
+ * @property {boolean} [wrapWithDirectory]
+ * @property {number} [timeout]
+ * @property {AbortSignal} [signal]
+ *
+ * @typedef {Object} AddAllInput
+ * @property {MultiFileInput} input
+ *
+ * @typedef {Object} AddInput
+ * @property {SingleFileInput} input
+ *
+ * @typedef {AddInput & AddOptions} AddQuery
+ * @typedef {AddAllInput & AddOptions} AddAllQuery
+ *
+ * @typedef {ArrayBuffer|ArrayBufferView|Blob|string|FileInput|RemoteIterable<ArrayBufferView|ArrayBuffer>} SingleFileInput
+ * @typedef {RemoteIterable<ArrayBuffer|ArrayBufferView|Blob|string|FileInput>} MultiFileInput
+ *
+ * @typedef {Object} FileInput
+ * @property {string} [path]
+ * @property {FileContent} content
+ * @property {Mode} [mode]
+ * @property {Time} [mtime]
+ *
+ * @typedef {ArrayBufferView|ArrayBuffer|Blob|string|RemoteIterable<ArrayBufferView>|RemoteIterable<ArrayBuffer>} FileContent
+ *
+ * @typedef {Object} AddedEntry
+ * @property {string} path
+ * @property {EncodedCID} cid
+ * @property {number} mode
+ * @property {UnixFSTime} mtime
+ * @property {number} size
+ *
+ * @typedef {Object} CatQuery
+ * @property {string} path
+ * @property {number} [offset]
+ * @property {number} [length]
+ *
+ * @typedef {Object} GetQuery
+ * @property {string} path
+ *
+ * @typedef {RemoteIterable<FileEntry>} GetResult
+ *
+ * @typedef {Object} FileEntry
+ * @property {string} path
+ * @property {RemoteIterable<Uint8Array>} content
+ * @property {Mode} [mode]
+ * @property {UnixFSTime} [mtime]
+ *
+ * @typedef {Object} LsQuery
+ * @property {string} path
+ *
+ * @typedef {RemoteIterable<LsEntry>} LsResult
+ *
+ * @typedef {Object} LsEntry
+ * @property {number} depth
+ * @property {string} name
+ * @property {string} path
+ * @property {number} size
+ * @property {EncodedCID} cid
+ * @property {FileType} type
+ * @property {Mode} mode
+ * @property {UnixFSTime} mtime
+ */
+
+/**
+ * @class
+ */
+class CoreService {
+  /**
+   *
+   * @param {IPFS} ipfs
+   */
+  constructor (ipfs) {
+    this.ipfs = ipfs
+  }
+
+  /**
+   * @typedef {Object} AddAllResult
+   * @property {RemoteIterable<AddedEntry>} data
+   * @property {Transferable[]} transfer
+
+   * @param {AddAllQuery} query
+   * @returns {AddAllResult}
+   */
+  addAll (query) {
```

Comment:
Does this need documenting in the readme?  It says it only supports `ipfs.add`.

Reply:
Right, I forgot to update docs after I incorporated changes from `addAll`.